src/applications/financial-status-report/components/otherExpenses/OtherExpensesSummary.jsx

```diff
@@ -43,29 +43,35 @@ const OtherExpensesSummary = ({
     ? 'Continue to review page'
     : 'Continue';
 
-  useEffect(
-    () => {
-      if (!gmtData?.isEligibleForStreamlined) return;
+  // Calculate Discretionary income as necessary
+  useEffect(() => {
+    if (!gmtData?.isEligibleForStreamlined) return;
 
-      const calculatedDiscretionaryIncome = calculateDiscretionaryIncome(data);
-      setFormData({
-        ...data,
-        gmtData: {
-          ...gmtData,
-          discretionaryBelow:
-            calculatedDiscretionaryIncome <
-            gmtData?.discretionaryIncomeThreshold,
-        },
-      });
-    },
-    // avoiding use of data since it changes so often
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [
-      otherExpenses,
-      gmtData?.isEligibleForStreamlined,
-      gmtData?.discretionaryIncomeThreshold,
-    ],
-  );
+    getMonthlyExpensesAPI(data).then(({ calculatedMonthlyExpenses }) => {
+      try {
+        const { totalMonthlyNetIncome } = getMonthlyIncome(data);
+        const calculatedDiscretionaryIncome =
+          totalMonthlyNetIncome - calculatedMonthlyExpenses;
+
```

Comment:
Could also convert this one to async/await?

Reply:
Same as above.